tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/effects/debugger_effects.ts

```diff
@@ -179,14 +249,38 @@ export class DebuggerEffects {
   /** @export */
   readonly loadData$: Observable<{}>;
 
-  /**
-   * When the debugger plugin is first loaded, request list of runs.
-   */
-  private onDebuggerLoaded() {
+  private onDebuggerDataPoll(): Observable<void> {
     return this.actions$.pipe(
-      // TODO(cais): Explore consolidating this effect with the greater
-      // webapp (in tensorboard/webapp), e.g., during PluginChanged actions.
       ofType(debuggerLoaded),
+      switchMap((action: Action) => {
+        return createTimedRepeater(
+          of(action),
+          this.store.select(getPollSilenceTimeMs).pipe(
+            map((pollSilenceTime) => {
+              return getCurrentPollingInterval(pollSilenceTime);
+            })
+          ),
+          this.actions$.pipe(ofType(debuggerUnloaded))
+        );
+      }),
+      tap(() => this.store.dispatch(debuggerDataPoll())),
+      map(() => void null)
+    );
+  }
+
+  private onCoreReload(): Observable<void> {
+    return this.actions$.pipe(
+      ofType(manualReload, reload),
+      tap(() => this.store.dispatch(debuggerDataPoll())),
```

Comment:
You can decide: Instead, I think the reducers should listen to above actions but I can see how it would be harder to reason when `lastDataPollOnsetTimeMs`. 

Rationale: this reload is really different user's intent that "poll" per say. It should be up to reducer to change the correct state based on actions.

Reply:
I prefer to keep this way because then all four ways of triggering a data poll in the dashboard will all go through a comment path in terms of triggering `lastDataPollOnsetTimeMs` change: via the action `debuggerDataPoll()`, which the reducer listens to.